src/TEdit/Editor/Plugins/RandomizerPlugin.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using TEdit.Terraria;
+using TEdit.ViewModel;
+
+
+namespace TEdit.Editor.Plugins
+{
+    public class RandomizerPlugin : BasePlugin
+    {
+        public RandomizerPlugin(WorldViewModel worldViewModel)
+            : base(worldViewModel)
+        {
+            Name = "Randomize All Blocks in the World";
+        }
+
+        public override void Execute()
+        {
+            if (_wvm.CurrentWorld == null)
+                return;
+
+            RandomizerPluginView view = new();
+            if (view.ShowDialog() == false)
+                return;
+
+            RandomizerSettings settings = new()
+            {
+                Seed = view.Seed
+            };
+
+            var mapping = GetRandomBlockMapping(settings);
+
+            for (int x = 0; x < _wvm.CurrentWorld.Size.X; x++)
+            {
+                for (int y = 0; y < _wvm.CurrentWorld.Size.Y; y++)
+                {
+                    Tile t = _wvm.CurrentWorld.Tiles[x, y];
+                    
+                    if (mapping.ContainsKey(t.Type))
+                    {
+                        t.Type = (ushort)mapping[t.Type];
+                    }
+                }
+            }
+            
+            _wvm.UpdateRenderRegion(new(0, 0, _wvm.CurrentWorld.Size.X, _wvm.CurrentWorld.Size.Y)); // Re-render map
+        }
+
+        private Dictionary<int, int> GetRandomBlockMapping(RandomizerSettings settings)
+        {
+            Dictionary<int, int> mapping = new();
+            Random rng = new Random(settings.Seed);
+
+            List<int> tiles = new List<int>(Terraria.World.TileBricks.Select((x) => x.Id));
```

Comment:
You will probably need to filter this list to _exclude_ framed tiles that require UV coordinates.

Reply:
I'm fairly certain that this bit of code makes sure there's no framed tiles in this list:

```
if (!curTile.IsFramed)
{
    TileBricks.Add(curTile);
    TileBricksMask.Add(curTile);
}
```

This is found in World.Settings.cs:289

I might be wrong though